lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -49,20 +49,20 @@ defmodule Mix.Tasks.Compile.Elixir do
   def run(args) do
     {opts, _, _} = OptionParser.parse(args, switches: @switches)
 
-    project = Mix.Project.config
     dest = Mix.Project.compile_path(project)
-    srcs = project[:elixirc_paths]
+    srcs = sources
     skip =
       case Keyword.get_values(opts, :elixirc_paths) do
         [] -> []
         ep -> srcs -- ep
       end
 
-    manifest = manifest()
     configs  = Mix.Project.config_files ++ Mix.Tasks.Compile.Erlang.manifests
-    force    = opts[:force] || Mix.Utils.stale?(configs, [manifest])
 
-    Mix.Compilers.Elixir.compile(manifest, srcs, skip, [:ex], dest, force, fn ->
+    keep_manifests = Map.take(manifest_map, srcs -- skip) |> Map.values
```

Comment:
This could probably be dried up with code in `compilers/elixir.ex`

This is necessary here so we don't force a recompile if another path hasn't been compiled since the change.

This may not be the behavior we want though. As an alternative we can error and say "you must compile the whole project after a configuration change"  or something like that.